hashintel/hash
# H-264: Implement type archival/unarchival in the frontend

## 🌟 What is the purpose of this PR?




This PR implements type archival in the FE for entity types.


## 🚫 Blocked by




- [x] merging https://github.com/hashintel/hash/pull/2876
- [x] merging https://github.com/hashintel/hash/pull/2888

## 🔍 What does this change?




- refactors the `TopContextBar` component to make it re-usable for archival across entity types, pages, and entities (entity archival not yet implemented)
- moves the archive banner into the `TopContextBar` to make it re-usable across entity types, pages and entities
- refactors `queryEntityTypesResolver` to accept an optional `includeArchived` param which will include archived types in the response
- refactors global `EntityTypesContext` to get all latest types including archived types

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:


### 📜 Does this require a change to the docs?




The changes in this PR:


### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:


## 🛡 What tests cover this?



None

## ❓ How to test this?



1. Checkout the branch, login to the FE
2. Create a type, and archive it via the context bar. Observe the type being removed from the sidebar
3. Unarchive the type via the context bar, and observe the type re-appearing in the sidebar

## 📹 Demo




https://github.com/hashintel/hash/assets/42802102/aacc6361-6f14-4c52-a3ae-8d9766613d26
